airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -565,7 +565,16 @@ def execute_async(self, context: Context):
 
     def convert_config_file_to_dict(self):
         """Converts passed config_file to dict format."""
-        config_file = self.config_file if self.config_file else os.environ.get(KUBE_CONFIG_ENV_VAR)
+        config_file = None
```

Comment:
I am not sure about the initial reason to convert the file into dictionary before creating the trigger, it may be to avoid copying the config file to the triggerer, where the pod is created on the worker using the sync hook and the waiting task is running on the triggerer and it uses the async hook.

> here is a risk that another is missing or new in the future would need "manual" fix like this

With this fix, we cover all options currently available to provide the configuration file, and yes, if we add a new one in the future, we must add it on the sync hook and in this method.

@VladaZakharova can you please explain what was the motivation to convert the config file to a dictionary before creating the trigger?

Reply:
Hi Team!
This was implemented to lighten the credential management: the config file was converted to dict to be passed to trigger and then hook to establish connection.

Reply:
what do you mean by `lighten the credential management` ?

the hook is not re instantiate at every run of the trigger ?

Reply:
We needed a way to pass config file to the trigger to create a client for kubernetes, but using file system to communicate with trigger was not a good solution. So then we added a possibility to pass all config file parameters as a dict.

Reply:
To respect the pattern mentioned by @raphaelauv, I will try loading the config file in the async hook, this should work where the triggerer is initiated once.

Reply:
Please mind that all FS operations are blocking side effects. It's violating asyncio contract and can cause additional error logs informing about blocking code.